sandbox-web/src/main/java/com/exadel/sandbox/team5/TagRestController.java

```diff
@@ -15,12 +16,22 @@
     private final TagService tagService;
 
     @GetMapping("/{id}")
-    public Tag getTag(@PathVariable Long id){
+    public Tag getTag(@PathVariable Long id) {
         return tagService.getById(id);
     }
 
     @GetMapping()
-    public List<Tag> getAll(){
+    public List<Tag> getAll() {
         return tagService.getAll();
     }
+
+    @PostMapping
+    public Tag save(@RequestBody Tag entity) {
+        return tagService.save(entity);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        tagService.delete(id);
```

Comment:
Please check case where tag is already used for some discounts

Reply:
remove delete method